compiler/code-gen/files/init-scripts.cpp

```diff
@@ -184,28 +182,24 @@ void InitScriptsCpp::compile(CodeGenerator &W) const {
   W << NL << StaticInit(all_functions) << NL;
 
   if (G->settings().is_static_lib_mode()) {
-    // only one main file is allowed for static lib mode
-    kphp_assert(main_file_ids.size() == 1);
-    W << LibGlobalVarsReset(main_file_ids.back()->main_function);
+    W << LibGlobalVarsReset(main_file_id->main_function);
     W << CloseFile();
     return;
   }
 
-  for (auto i : main_file_ids) {
-    W << RunFunction(i->main_function) << NL;
-    W << GlobalResetFunction(i->main_function) << NL;
-  }
+  W << RunFunction(main_file_id->main_function) << NL;
+  W << GlobalResetFunction(main_file_id->main_function) << NL;
 
   FunctionSignatureGenerator(W) << "void init_php_scripts() " << BEGIN;
 
-  for (auto i : main_file_ids) {
-    W << FunctionName(i->main_function) << "$global_reset();" << NL;
+  W << FunctionName(main_file_id->main_function) << "$global_reset();" << NL;
 
-    W << "set_script (" <<
-      "\"@" << i->short_file_name << "\", " <<
-      FunctionName(i->main_function) << "$run, " <<
-      FunctionName(i->main_function) << "$global_reset);" << NL;
-  }
+  // TODO: do we need a scripts map if it contains only 1 script?
+  // A further cleanup may be possible.
```

Comment:
Let's do it right now. It's easy, just go to `server/php-script.cpp` and 
1) replace `std::map` with single `script_t *`
2) drop `const char *name` from all functions

Reply:
Done.